node/src/components/block_executor.rs

```diff
@@ -132,7 +134,13 @@ impl BlockExecutor {
         effect_builder: EffectBuilder<REv>,
         finalized_block: FinalizedBlock,
     ) -> Effects<Event> {
-        let deploy_hashes = SmallVec::from_slice(finalized_block.proto_block().deploys());
+        let deploy_hashes = finalized_block
+            .proto_block()
+            .deploys()
+            .iter()
+            .map(|hash| **hash)
+            .collect::<Vec<_>>();
+        let deploy_hashes = SmallVec::from_slice(&deploy_hashes[..]);
```

Comment:
Is this code allocating twice vs once previously?

Reply:
I think so. You should be able to directly `collect()` into a `SmallVec`.